File: tools/seed_test_vehicle.py

import psycopg2
import bcrypt
import os

# DB Config
DB_PARAMS = {
    "dbname": "safedrive_apps",
    "user": "safedrive_user",
    "password": "safedrive",
    "host": "localhost",
    "port": "5432"
}

def seed_data():
    try:
        conn = psycopg2.connect(**DB_PARAMS)
        cur = conn.cursor()
        
        email = "testuser3@example.com"
        password = "password123"
        hashed = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
        
        # 1. Insert User
        print(f"Creating User: {email} ...")
        # Check if exists
        cur.execute("SELECT id FROM users WHERE email = %s", (email,))
        res = cur.fetchone()
        
        if res:
            user_id = res[0]
            print(f"  Result: User already exists (ID: {user_id})")
        else:
            cur.execute("""
                INSERT INTO users (name, email, password, mobile, wallet_balance)
                VALUES (%s, %s, %s, %s, %s)
                RETURNING id
            """, ("Test User 3", email, hashed, "9999999999", 500.0))
            user_id = cur.fetchone()[0]
            print(f"  Result: Created User ID: {user_id}")
            
        # 2. Insert Vehicle
        plate = "MH12TEST03"
        print(f"Adding Vehicle: {plate} ...")
        
        cur.execute("SELECT id FROM vehicles WHERE plate_number = %s", (plate,))
        if cur.fetchone():
             print(f"  Result: Vehicle already exists.")
        else:
            cur.execute("""
                INSERT INTO vehicles (user_id, plate_number, vehicle_type, is_primary)
                VALUES (%s, %s, %s, %s)
            """, (user_id, plate, "Car", True))
            print(f"  Result: Vehicle Added.")
            
        conn.commit()
        cur.close()
        conn.close()
        print("\nSUCCESS! Login with:")
        print(f"Email: {email}")
        print(f"Pass:  {password}")
        
    except Exception as e:
        print(f"\nERROR: {e}")

if __name__ == "__main__":
    seed_data()

File: SafeDrive/backend_server/seed_test_vehicle.js

const { Client } = require('pg');
const bcrypt = require('bcryptjs');

// Config
const DB_CONFIG = {
    user: 'postgres',
    host: 'localhost',
    database: 'safedrive_apps',
    password: 'Siraj.9892',
    port: 5432,
};

async function seed() {
    const client = new Client(DB_CONFIG);
    try {
        await client.connect();
        console.log("Connected to DB...");

        const email = "testuser3@example.com";
        const password = "password123";
        const salt = await bcrypt.genSalt(10);
        const hashed = await bcrypt.hash(password, salt);

        // 1. Insert User
        console.log(`Creating User: ${email}...`);

        let res = await client.query('SELECT user_id FROM users WHERE email = $1', [email]);
        let userId;

        if (res.rows.length > 0) {
            userId = res.rows[0].user_id;
            console.log(`  User already exists (ID: ${userId})`);
        } else {
            res = await client.query(
                `INSERT INTO users (name, email, password_hash, mobile, wallet_balance)
                 VALUES ($1, $2, $3, $4, $5) RETURNING user_id`,
                ["Test User 3", email, hashed, "9999999999", 500.0]
            );
            userId = res.rows[0].user_id;
            console.log(`  Created User ID: ${userId}`);
        }

        // 2. Insert Vehicle
        const plate = "MH12TEST03";
        console.log(`Adding Vehicle: ${plate}...`);

        res = await client.query('SELECT vehicle_id FROM vehicles WHERE plate_number = $1', [plate]);
        if (res.rows.length > 0) {
            console.log(`  Vehicle already exists.`);
        } else {
            await client.query(
                `INSERT INTO vehicles (user_id, plate_number, vehicle_type, is_primary)
                 VALUES ($1, $2, $3, $4)`,
                [userId, plate, "Car", true]
            );
            console.log(`  Vehicle Added.`);
        }

        console.log("\nSUCCESS! Login with:");
        console.log(`Email: ${email}`);
        console.log(`Pass:  ${password}`);

    } catch (err) {
        console.error("ERROR:", err);
    } finally {
        await client.end();
    }
}

seed();
